src/h3lib/lib/h3UniEdge.c

```diff
@@ -229,14 +242,25 @@ void H3_EXPORT(getH3UnidirectionalEdgeBoundary)(H3Index edge, GeoBoundary* gb) {
 
     int k = 0;
     for (int i = 0; i < origin.numVerts; i++) {
-        for (int j = 0; j < destination.numVerts; j++) {
-            if (geoAlmostEqualThreshold(&origin.verts[i], &destination.verts[j],
-                                        0.000001)) {
-                gb->verts[k].lat = origin.verts[i].lat;
-                gb->verts[k].lon = origin.verts[i].lon;
+        if (_hasMatchingVertex(&origin.verts[i], &destination)) {
+            // If we are on vertex 0, we need to handle the case where it's the
+            // end of the edge, not the beginning.
+            // TODO: This is fine for this limited case, but it might be
+            // better/cleaner to use a VertexGraph for this
+            if (i == 0 &&
+                !_hasMatchingVertex(&origin.verts[i + 1], &destination)) {
+                postponedVertex = origin.verts[i];
+                hasPostponedVertex = true;
```

Comment:
Is zero a valid vertex? If not, you can just check that `postponedVertex != 0` instead of maintaining a separate flag.

Reply:
`postponedVertex` is a `GeoCoord` struct, and unfortunately, as @isaacbrodsky points out, it's valid to have all the fields in the struct initialized to `0`. So I don't know of another way to check whether `postponedVertex` has been set or not, hence the flag. Open to suggestions.